src/EnergyPlus/SimAirServingZones.cc

```diff
@@ -1900,6 +1903,29 @@ namespace SimAirServingZones {
 
 			} // end of AirLoop loop
 
+			// Check whether there are Central Heating Coils in the Primary Air System
+			for( AirLoopNum = 1; AirLoopNum <= NumPrimaryAirSys; ++AirLoopNum ) {
+
+				FoundCentralHeatCoil = false;
+
+				for( BranchNum = 1; BranchNum <= PrimaryAirSystem( AirLoopNum ).NumBranches; ++BranchNum ) {
+
+					for( CompNum = 1; CompNum <= PrimaryAirSystem( AirLoopNum ).Branch( BranchNum ).TotalComponents; ++CompNum ) {
+
+						CompTypeNum = PrimaryAirSystem( AirLoopNum ).Branch( BranchNum ).Comp( CompNum ).CompType_Num;
+
+						if( CompTypeNum == WaterCoil_SimpleHeat || CompTypeNum == Coil_ElectricHeat || CompTypeNum == Coil_GasHeat ) {
+							FoundCentralHeatCoil = true;
```

Comment:
@rongpengzhang This could maybe be combined with the above (starting line 1874) for loop, but it would require a little refactoring. The quicker change is to at least break from these for loops once `FoundCentralHeatCoil` is found. I don't really like `goto` like the above for loop uses, so i'd probably put `FoundCentralHeatCoil` in the for loop checks themselves. For example,

```cpp
for( BranchNum = 1; ! FoundCentralHeatCoil && BranchNum <= PrimaryAirSystem( AirLoopNum ).NumBranches; ++BranchNum ) {
  for( CompNum = 1; ! FoundCentralHeatCoil && CompNum <= PrimaryAirSystem( AirLoopNum ).Branch( BranchNum ).TotalComponents; ++CompNum ) {
    CompTypeNum = PrimaryAirSystem( AirLoopNum ).Branch( BranchNum ).Comp( CompNum ).CompType_Num;
    if( CompTypeNum == WaterCoil_SimpleHeat || CompTypeNum == Coil_ElectricHeat || CompTypeNum == Coil_GasHeat ) {
      FoundCentralHeatCoil = true;
    }
  } // end of component loop
} // end of Branch loop
```

This code should work to break early but definitely test it.